packages/flutter/lib/src/material/shaders/ink_sparkle.frag

```diff
@@ -11,22 +11,19 @@ precision highp float;
 // TODO(antrob): Put these in a more logical order (e.g. separate consts vs varying, etc)
 
 layout(location = 0) uniform vec4 u_color;
-layout(location = 1) uniform float u_alpha;
-layout(location = 2) uniform vec4 u_sparkle_color;
-layout(location = 3) uniform float u_sparkle_alpha;
-layout(location = 4) uniform float u_blur;
-layout(location = 5) uniform vec2 u_center;
-layout(location = 6) uniform float u_radius_scale;
-layout(location = 7) uniform float u_max_radius;
-layout(location = 8) uniform vec2 u_resolution_scale;
-layout(location = 9) uniform vec2 u_noise_scale;
-layout(location = 10) uniform float u_noise_phase;
-layout(location = 11) uniform vec2 u_circle1;
-layout(location = 12) uniform vec2 u_circle2;
-layout(location = 13) uniform vec2 u_circle3;
-layout(location = 14) uniform vec2 u_rotation1;
-layout(location = 15) uniform vec2 u_rotation2;
-layout(location = 16) uniform vec2 u_rotation3;
+// u_alpha, u_sparkle_alpha, u_blur, u_radius_scale
+layout(location = 1) uniform vec4 u_composite_1;
```

Comment:
Drive by comment: The brittle-ness of specifying locations explicitly and then matching the same in Dart code seems egregious. The compiler auto-generates these and produces metdata. The C++ side uses this metadata for example. Perhaps we could do the same for Dart?

Reply:
If you want to be the next lemming to walk off the cliff of getting Dart code gen to work, be my guest, however I will not be working on this 😄

Reply:
I was thinking of something along the lines of JSON file that is read at runtime that contains a mapping between uniform names and their locations.

But, I must admit I haven't been burned by attempted to get Dart code-gen to work... yet. Perhaps why we can document why it sucks. We are generating code in C++ and I'd say it has worked... fine? Are we talking about specific challenges with Dart that will make it way harder than the engine C++ setup?